torchdynamo/eval_frame.py

```diff
@@ -364,6 +369,105 @@ def toy_example(a, b):
     )
 
 
+def evaluate(
+    f: Callable,
+    backends: Set[str] = [None, "inductor"],
+    inputs: List[Any] = [],
+    *,
+    warmups: List[Any] = [],
+    file=None,
+    nopython=False,
+):
+    assert f is not None, "Must provide a callabke to evaluate"
+    assert len(backends) > 0, "Must provide at least 1 backend"
+    assert len(inputs) > 0, "Must provide at least 1 input"
+
+    import cProfile
+    from pstats import Stats
+
+    backends = set(backends)
+
+    print(f"Starting evaluation of {len(backends)} backends: {backends}", file=file)
+
+    assert isinstance(backends, Set), "Backends must be a set"
+
+    result_lookup_table = []
+    for input_idx in range(0, len(inputs)):
+        result_lookup_table.append({})
+
+    for backend in backends:
+        if backend is None:
+            backend = "No Dynamo"
+
+        # TODO(voz): If backend is inductor, get data from the inductor trace util
+
+        print(f"EVALUATING {backend}", file=file)
+        torchdynamo.reset()
+        backend_fn = f
+        # A None backend indicates a run without TorchDynamo
+        if backend != "No Dynamo":
+            backend_fn = torchdynamo.optimize(backend, nopython=nopython)(f)
+
+        first_input = inputs[0]
+        (
+            explanation,
+            _,
+            _,
+            _,
+            _,
+        ) = torchdynamo.explain(f, first_input)
+        print(f"EXPLAIN FOR {backend}: {explanation}", file=file)
+        torchdynamo.reset()
+        try:
+            for warmup in warmups:
+                backend_fn(warmup)
+            if len(warmups) > 0:
+                print(f"WARMUP FINISHED FOR {backend}", file=file)
+                
+        except Exception as e:
+            print(f"WARMUP FAILED FOR {backend}", file=file)
+            raise e
+
+        try:
+            pr = cProfile.Profile()
+            pr.enable()
+
+            start = time.time()
+            for input_idx in range(0, len(inputs)):
+                input = inputs[input_idx]
+                result = backend_fn(input)
```

Comment:
Same as previous comment
```suggestion
                result = backend_fn(*input)
```

Reply:
Same as previous comment